route_guide/route_guide_client.cc

```diff
@@ -23,238 +23,324 @@
   RouteGuideClient(const std::string& db, Client* client) : client_(client) {
     routeguide::ParseDb(db, &feature_list_);
   }
 
-  void GetFeature() {
-    Point point;
-    Feature feature;
-    point = MakePoint(409146138, -746188906);
-    GetOneFeature(point, &feature);
-    point = MakePoint(0, 0);
-    GetOneFeature(point, &feature);
-  }
-
-  void ListFeatures() {
-    Rectangle rect;
-    Feature feature;
-
-    rect.mutable_lo()->set_latitude(400000000);
-    rect.mutable_lo()->set_longitude(-750000000);
-    rect.mutable_hi()->set_latitude(420000000);
-    rect.mutable_hi()->set_longitude(-730000000);
-    std::cout << "Looking for features between 40, -75 and 42, -73"
-              << std::endl;
-
-    Notification<Status> done;
-    client_->Call<RouteGuide, Rectangle, Stream<Feature>>(
-        "ListFeatures",
-        &rect,
-        [this](auto* call, auto&& feature) {
-          if (feature) {
-            std::cout << "Found feature called "
-                      << feature->name() << " at "
-                      << feature->location().latitude()/kCoordFactor_ << ", "
-                      << feature->location().longitude()/kCoordFactor_ << std::endl;
-          } else {
-            call->Finish();
-          }
-        },
-        [&](auto*, const Status& status) {
-          done.Notify(status);
-        });
+ public:
+  // The goal of GetFeature is just to implement async unary rpc call.
+  // Client sends to the server the request which contains Point's coordi-
+  // nates (longitude, latitude). The server checks received Point's coordi-
+  // nates, if database has the specific address which corresponds to Point's
+  // coordinates - the server will response with it address. GetFeature
+  // returns lambda.
+  auto GetFeature(long latitude, long longitude) {
+    // The idea of Data structure is just to use it in the .context() callback
+    // in order to have the way of injecting some data on a stack, or as
+    // a part of the eventual data structure and manipulate with this data in
+    // multiple callbacks (.start(), .fail(), .ready(), .body(), etc).
+    struct Data {
+      Point point;      // Request from the client
+      Feature feature;  // Response from the server
+    };
 
-    Status status = done.Wait();
-    if (status.ok()) {
-      std::cout << "ListFeatures rpc succeeded." << std::endl;
-    } else {
-      std::cout << "ListFeatures rpc failed." << std::endl;
-    }
+    return [latitude, longitude, this]() {
+      // Call() function gets the rpc name and the host name as parameters
+      // and returns Composable structure which is composed with Handler()
+      // call using overloaded operator "|" in order to handle calls from
+      // server. Actually Handler() returns Composable structure too, so it
+      // can be composed with other tasks. (You can check Composable
+      // structure's implementation in the stout/grpc directory in handler.h)
```

Comment:
Rather than diving into implementation details, how about we only focus on the usage details:
```suggestion
      // Call() is an `Eventual` (a concept from `stout-eventuals`) and can 
      // be composed into an asynchronous pipeline with other Eventuals. 
      // In this example we use the "|" operator to compose it with a 
      // `Client::Handler()`. The handler in turn could be composed also. 
      // For more information about Eventuals, see: 
      //   https://github.com/3rdparty/stout-eventuals.
```